mono/metadata/threadpool-ms-io.c

```diff
@@ -62,23 +68,42 @@ enum {
 	AIO_OP_LAST
 };
 
+typedef enum {
+	UPDATE_EMPTY = 0,
+	UPDATE_ADD,
+	UPDATE_REMOVE_SOCKET,
+	UPDATE_REMOVE_DOMAIN,
+} ThreadPoolIOUpdateType;
+
 typedef struct {
 	gint fd;
 	MonoSocketAsyncResult *sockares;
-} ThreadPoolIOUpdate;
+} ThreadPoolIOUpdate_Add;
```

Comment:
That would make the whole point of an update queue kind of pointless.

I worked around by registering it as a conservative root. Not the right solution, but one that works.